include/atomic_queue/moodycamel.h

```diff
@@ -15,29 +15,53 @@ namespace atomic_queue {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-template<class T, unsigned Capacity, class Queue>
-struct MoodyCamelAdapter : Queue {
-    MoodyCamelAdapter()
-        : Queue(Capacity) {}
+template<class T, unsigned Capacity>
+struct MoodyCamelQueue : moodycamel::ConcurrentQueue<T> {
+    MoodyCamelQueue()
+        : moodycamel::ConcurrentQueue<T>(Capacity) {
+        ctok() = moodycamel::ConcurrentQueue<T>::consumer_token_t(*this);  // update instance
+    }
 
     void push(T element) {
-        while(!this->try_enqueue(element))
+        // TODO: To be able to use try_enqueue properly, particularly with tokens,
+        // requires the number of producer and consumer threads to be passed to the
+        // constructor
```

Comment:
Would you like to pass that number into the constructor then?

Reply:
That would be ideal but I didn't want to change your interface/API.

Reply:
Feel free to change the implementation and the API of the benchmark, please.

If not, I will try to route elegantly the consumer/producer number into your queue constructor by the end of this week.

Reply:
Alright I've pushed a change that passes the producer thread count and adds tokens. I changed the moodycamel, Boost, and Xenium adapters, but AtomicQueuea and TBB probably need new adapters.

I can't compile on my platform (Windows with no Boost, TBB, etc.) so my changes may need a bit of work before merging.